text/accumulation.tex

```diff
@@ -30,118 +34,206 @@ \subsection{Preimage Integration}
 \begin{align}
-  M\colon\left\{\ \begin{aligned}
-    \N_S &\to \lseq\mathbb{O}\rseq \\
-    \displaystyle s &\mapsto \left[ \ltup\begin{alignedat}{3}
-        \is{o&}{\mathbf{r}_o}\ts\is{&l&}{\mathbf{r}_p}\ts\\
-        \is{a&}{w_o}\ts\is{&k&}{(w_s)_h}
-    \end{alignedat}\rtup
-      \,\middle\vert\ 
-    \begin{aligned}
-      w &\in \mathbf{W},\\
-      \mathbf{r} &\in w_\mathbf{r},\\
-      \mathbf{r}_s &= s
-    \end{aligned}\ \right]
-  \end{aligned}\right.
+  \mathbf{W}^! &\equiv [ w \mid w \orderedin \mathbf{W}, (w_x)_p = \none \wedge w_\wr¬srlookup = \{\} ] \\
+  \mathbf{W}^Q &\equiv E(\sq{
+    D(w) \mid
+    w \orderedin \mathbf{W},
+    (w_x)_p \ne \none \vee w_\wr¬srlookup \ne \{\}
+  }, \wideparen{\accumulated})\!\!\\
+  D(w) &\equiv (w, (w_x)_p \cup \keys{w_\wr¬srlookup})
 \end{align}
 
-\subsection{Invocation}
+We must define the two dependent functions. Firstly, the mapping function $\srmap$ which builds a dictionary of work-package hashes to segment-roots from a set of work-reports. Formally:
+\begin{equation}
+  \srmap\colon\left\{\begin{aligned}
+    \{\mathbb{W}\} &\to \dict{\H}{\H}\\
+    \mathbf{w} &\mapsto \{ ((w_s)_h \mapsto (w_s)_e) \mid w \in \mathbf{w} \}
+  \end{aligned}\right.
+\end{equation}
 
-Within this section, we define $A$, the function which conducts the accumulation of a single service. Formally speaking, $A$ assumes omnipresence of timeslot $\mathbf{H}_t$ and some prior state components $\delta^\dagger$, $\nu$, $\mathbf{W}_\mathbf{d}$, and takes as specific arguments the service index $s \in \mathbf{S}$ (from which it may derive the wrangled results $M(s)$ and gas limit $G(s)$) and yields values for $\delta^\ddagger[s]$ and staging assignments into $\varphi$, $\iota$ together with a series of lookup solicitations/forgets, a series of deferred transfers and $\mathbf{C}$ mapping from service index to \textsc{Beefy} commitment hashes.
+Next, the queue-editing function $E$, which is essentially a mutator function for items such as those of $\ready$, parameterized by a dictionary similar to the items in $\accumulated$. It is used to update queues of work-reports when some of them are accumulated. Functionally, it removes all entries whose work-report's hash is in the parameter's keys, removes any dependencies which appear in said set and, crucially, removes any entries whose segment-root correspondences diverge from those in the dictionary. Formally:
+\begin{equation}
+  E\colon\left\{\begin{aligned}
+      &(\seq{(\mathbb{W}, \{\H\})}, \dict{\H}{\H}) \to \seq{(\mathbb{W}, \{\H\})} \\
+    &(\mathbf{r}, \mathbf{x}) \mapsto \left[
+      (w, \mathbf{d} \setminus \keys{\mathbf{x}})
+      \,\middle\vert\,
+      \left\{\,\begin{aligned}
+        &(w, \mathbf{d}) \orderedin \mathbf{r} ,\\
+        &\mathcal{H}(\se(w)) \not\in \keys{\mathbf{x}},\\
```

Comment:
Think instead of H(w) you want (w_s)_h? AIUI w is a work report and you want the WP hash.